source/Octopus.Tentacle.Tests/Integration/StartScriptCommandV2Builder.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Scripts;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    public class StartScriptCommandV2Builder
+    {
+        readonly List<ScriptFile> files = new List<ScriptFile>();
+        readonly List<string> arguments = new List<string>();
+        readonly Dictionary<ScriptType, string> additionalScripts = new Dictionary<ScriptType, string>();
+        StringBuilder scriptBody = new StringBuilder(string.Empty);
+        ScriptIsolationLevel isolation = ScriptIsolationLevel.FullIsolation;
+        TimeSpan scriptIsolationMutexTimeout = ScriptIsolationMutex.NoTimeout;
+        string scriptIsolationMutexName = nameof(RunningScript);
+        string taskId = Guid.NewGuid().ToString();
+        ScriptTicket scriptTicket = new ScriptTicket(Guid.NewGuid().ToString());
+        TimeSpan durationStartScriptCanWaitForScriptToFinish = TimeSpan.Zero;
+
+        public StartScriptCommandV2Builder WithScriptBody(string scriptBody)
+        {
+            this.scriptBody = new StringBuilder(scriptBody);
+            return this;
+        }
+
+        public StartScriptCommandV2Builder WithScriptBody(string windowsScript, string bashScript)
+        {
+            this.scriptBody = new StringBuilder(PlatformDetection.IsRunningOnWindows ? windowsScript : bashScript);
+            return this;
+        }
+
+        public StartScriptCommandV2Builder WithReplacementInScriptBody(string oldValue, string newValue)
```

Comment:
Does this method need to exist here?

If it does, I wonder if it is worth a comment saying this needs to be called after the script is set. It feels like a trap since you could do:

```
WithReplacementInScriptBody("foo", "bar")
.WithScriptBody("foo")
```

And be left with foo.

Reply:
This is a copy of the original builder